pkg/namespacestore/validation.go

```diff
@@ -7,40 +7,47 @@ import (
 	"strings"
 
 	nbv1 "github.com/noobaa/noobaa-operator/v5/pkg/apis/noobaa/v1alpha1"
+	"github.com/noobaa/noobaa-operator/v5/pkg/system"
 	"github.com/noobaa/noobaa-operator/v5/pkg/util"
 )
 
+// const configuration values for the validation checks
 const (
-	defaultEndPointURI = "https://127.0.0.1:6443"
+	defaultEndPointURI     = "https://127.0.0.1:6443"
+	MaximumMountPathLength = 63
 )
 
 // ValidateNamespaceStore validates namespacestore configuration
 func ValidateNamespaceStore(nsStore *nbv1.NamespaceStore) error {
+	if err := ValidateNSEmptySecretName(*nsStore); err != nil {
+		return err
+	}
 	switch nsStore.Spec.Type {
 
 	case nbv1.NSStoreTypeNSFS:
-		return validateNsStoreNSFS(nsStore)
+		return ValidateNsStoreNSFS(nsStore)
 
 	case nbv1.NSStoreTypeAWSS3:
 		return nil
 
 	case nbv1.NSStoreTypeS3Compatible:
-		return validateNsStoreS3Compatible(nsStore)
+		return ValidateNsStoreS3Compatible(nsStore)
 
 	case nbv1.NSStoreTypeIBMCos:
-		return validateNsStoreIBMCos(nsStore)
+		return ValidateNsStoreIBMCos(nsStore)
 
 	case nbv1.NSStoreTypeAzureBlob:
 		return nil
 
 	default:
-		return util.NewPersistentError("ConfigurationError",
```

Comment:
by removing the persistent error, the places that already call this function will keep reconciling, re-check that change if this is the desired behavior

Reply:
correct, I fixed it by triggering a `NewPersistentError` in the places that already call this function (BTW its called only twice, once in `namespacestore.go` and if the validations fails it will `log.Fatalf()`, and the second place is in the reconciler which then I trigger a persistent error).

Reply:
since validation.go under backingstore is a new file we currently don't have this issue in backingstore, why won't we change the equivalent places in backingstore.go and backingstore reconciler to call to correct validations in validation file?

Reply:
you are right, but I think this change is out of scope for this PR, the end goal is to implement what you described.

Reply:
let's add the backingstore reconciler validations as part of this refactor